android/src/main/java/io/branch/rnbranch/RNBranchModule.java

```diff
@@ -211,13 +215,21 @@ private void generateLocalBroadcast(JSONObject referringParams,
         Branch.sessionBuilder(reactActivity).withCallback(referralInitListener).withData(uri).init();
     }
 
+    public static void setIntent(Intent intent, @Nonnull Activity activity) {
```

Comment:
can we rename this to `onNewIntent`, so it's clearer that it needs to be called form `onNewIntent` only. Also I believe onNewIntent will never return `null`, so we can mark the argument as `@NonNull`.